services/src/main/java/edu/unc/lib/dl/cdr/services/rest/modify/MoveObjectsController.java

```diff
@@ -46,32 +50,70 @@
+	
+	private final Map<String, MoveRequest> activeMoveRequests;
+	
+	public MoveObjectsController() {
+		activeMoveRequests = new HashMap<>();
+	}
 
 	@RequestMapping(value = "edit/move", method = RequestMethod.POST)
 	public @ResponseBody
-	String moveObjects(@RequestBody MoveRequest moveRequest, Model model, HttpServletRequest request,
-			HttpServletResponse response) {
+	Map<String, Object> moveObjects(@RequestBody MoveRequest moveRequest, HttpServletResponse response) {
+		Map<String, Object> results = new HashMap<>();
 		// Validate that the request contains the newPath and ids fields.
-		if (moveRequest == null || moveRequest.ids == null || moveRequest.ids.size() == 0
-				|| moveRequest.getNewParent() == null || moveRequest.getNewParent().length() == 0) {
+		if (moveRequest == null || moveRequest.moved == null || moveRequest.moved.size() == 0
+				|| moveRequest.getDestination() == null || moveRequest.getDestination().length() == 0) {
 			response.setStatus(400);
-			return "{\"error\": \"Request must provide a newParent destination and a list of ids\"}";
+			results.put("error", "Request must provide a destination destination and a list of ids");
+			return results;
 		}
+		
+		moveRequest.setUser(GroupsThreadStore.getUsername());
 
-		List<PID> pids = new ArrayList<PID>(moveRequest.getIds().size());
-		for (String id : moveRequest.getIds())
-			pids.add(new PID(id));
-		PID parent = new PID(moveRequest.getNewParent());
-
-		Thread moveThread = new Thread(new MoveRunnable(pids, parent, GroupsThreadStore.getUsername(),
-				GroupsThreadStore.getGroups()));
+		Thread moveThread = new Thread(new MoveRunnable(moveRequest, GroupsThreadStore.getGroups()));
 		log.info("User {} is starting move operation of {} objects to destination {}",
-				new Object[] { GroupsThreadStore.getUsername(), pids.size(), parent });
+				new Object[] { GroupsThreadStore.getUsername(), moveRequest.moved.size(), moveRequest.getDestination()});
 		moveThread.start();
 
-		response.setStatus(204);
-		return "{\"message\": \"Operation to move " + pids.size() + " objects into container " + parent.getPid()
-				+ " has begun\"}";
+		response.setStatus(200);
+		
+		results.put("id", moveRequest.id);
+		results.put("message", "Operation to move " + moveRequest.moved.size() + " objects into container "
+				+ moveRequest.getDestination() + " has begun");
+		return results;
+	}
+
+	@RequestMapping(value = "listMoves", method = RequestMethod.GET)
+	public @ResponseBody
+	Object listMoves() {
+		Map<String, Boolean> moves = new HashMap<>(this.activeMoveRequests.size());
+		
+		Iterator<Entry<String, MoveRequest>> moveIt = activeMoveRequests.entrySet().iterator();
+		while (moveIt.hasNext()) {
+			Entry<String, MoveRequest> move = moveIt.next();
+			long finishedAt = move.getValue().finishedAt;
+			if (finishedAt == -1 || finishedAt + TIME_TO_LIVE_AFTER_FINISH > System.currentTimeMillis()) {
+				moves.put(move.getKey(), finishedAt == -1);
```

Comment:
having a value of false means that the job is not active, but the server is still reporting on it.  It prevents the client from looking up info about the job after it finishes but before it has been cleaned up